features/0586-sign-attachment/README.md

```diff
@@ -0,0 +1,219 @@
+
+The attachment may be signed. if so, a `jws` signature will be present in the `data` attribute of the attachment. The signer may use this signature as part of their evaluation, but the signature they produce will be across the payload only.
+
+
+### Signature Response
+
+This message represents the response from the *signer* to the *coordinator* containing the response to the "please sign this" request.  The response will contain the requested signature(s), or an indicator that the signer can't or won't sign the request. The request can contain multiple attachments, so the response can contain multiple signatures (or refusals).  The response will *not* include the attachment(s) from the request, but each signature will include the `@id` of the associated attachment.
+
+```jsonc
+{
+  "@type": "http://didcomm.org/sign-attachment/%VER/signature-response",
+  "@id": "ece30ed1-96f8-44c9-95cf-b274288009dc",
+  "~thread": {
+    "thid": "fce30ed1-96f8-44c9-95cf-b274288009dc"
+  },
+  "signature_response": [{
+      "message_id": "143c458d-1b1c-40c7-ab85-4d16808ddf0a",
+      "context": "did:sov",
+      "method": "add-signature",
+      "signer_goal_code": "transaction.endorse" # or "transaction.refuse",
+      "signature_type": "<requested signature type>",
+      "signature": { ... structure determined by signature type ... }
+  }]
+}
+```
+**signature_response** - An array of signature responses. This should correlate to the number of requested signatures.
+- **message_id**: Corresponds to the `@id` of the message 
+- **context**: same as in request
+- **method**: same as in request
+- **signer_goal_code**: the same as the `signer_goal_code` of the request. If the signature is refused, a different suitable goal code should be used. 
+- **signature_type"**: requested signature type
+- **signature**: a structure, determined by the signature type.
+
+
+In the case of an error, a problem report *should* be returned to the requesting agent instead of an endorsement response message.  (However if the endorser declines to endorse, they will return a response with that goal code.)
+
+### Signature Present
+
+This message presents gathered signatures to the message recipient, along with the attachment that was signed. This allows a witness to receive and verify the provided signatures.
+
+```jsonc
+{
+  "@type": "http://didcomm.org/sign-attachment/%VER/signature-present",
+  "@id": "ece30ed1-96f8-44c9-95cf-b274288009dc",
+  "~thread": {
+    "thid": "fce30ed1-96f8-44c9-95cf-b274288009dc"
+  },
+  "signatures": [{
+      "message_id": "143c458d-1b1c-40c7-ab85-4d16808ddf0a",
+      "context": "did:sov",
+      "method": "add-signature",
+      "signer_goal_code": "transaction.endorse" # or "transaction.refuse",
+      "signature_type": "<requested signature type>",
+      "signature": { ... structure determined by signature type ... }
+  }],
+  "messages~attach": [{
```

Comment:
Shouldn't this be `~attach` to match the request? Although I would prefer just calling it `messages`.

Reply:
I moved to the attachment append method (as opposed to inline) partially as an experiment to see what it would look like. We are considering only appended attachments in DIDComm V2.

Reply:
But - I did make it consistent.